From 36956756e27e3ed6141a8b2bb5655c653d503e3f Mon Sep 17 00:00:00 2001
From: Ram Kumar Dwivedi <quic_rdwivedi@quicinc.com>
Date: Mon, 11 Nov 2024 12:55:08 +0530
Subject: [PATCH] ufs: Configure UFS driver for NordAU SoD bringup

Configure UFS driver for NordAU RUMI:
1. Set max gear to Gear 4
2. Set max mode to Rate B
3. Enable all LPM capabilities
4. Enable power mode change during hba probe
5. Enable 64 bit DMA mask.

Change-Id: I82f84f7308c48f054e810b8434cd5a1ce8fdef42
Signed-off-by: Ram Kumar Dwivedi <quic_rdwivedi@quicinc.com>
---
 drivers/ufs/core/ufshcd.c        |  7 +------
 drivers/ufs/host/ufs-qcom.c      |  8 +-------
 drivers/ufs/host/ufs-qcom.h      |  2 +-
 drivers/ufs/host/ufshcd-pltfrm.c | 14 +++++++-------
 4 files changed, 10 insertions(+), 21 deletions(-)

diff --git a/drivers/ufs/core/ufshcd.c b/drivers/ufs/core/ufshcd.c
index f5cd709fbdb1..45c40d75255a 100644
--- a/drivers/ufs/core/ufshcd.c
+++ b/drivers/ufs/core/ufshcd.c
@@ -4528,7 +4528,7 @@ static int ufshcd_get_max_pwr_mode(struct ufs_hba *hba)
 		pwr_info->pwr_tx = FAST_MODE;
 		pwr_info->pwr_rx = FAST_MODE;
 	}
-	pwr_info->hs_rate = PA_HS_MODE_A;
+	pwr_info->hs_rate = PA_HS_MODE_B;
 
 	/* Get the connected lane count */
 	ufshcd_dme_get(hba, UIC_ARG_MIB(PA_CONNECTEDRXDATALANES),
@@ -8849,9 +8849,6 @@ static int ufshcd_device_init(struct ufs_hba *hba, bool init_dev_params)
 
 	ufshcd_set_timestamp_attr(hba);
 
-	dev_err(hba->dev, "%s: Power mode change disabled\n",__func__);
-	return 0;
-
 	/* Gear up to HS gear if supported */
 	if (hba->max_pwr_info.is_valid) {
 		/*
@@ -10355,12 +10352,10 @@ EXPORT_SYMBOL_GPL(ufshcd_dealloc_host);
  */
 static int ufshcd_set_dma_mask(struct ufs_hba *hba)
 {
-/*
 	if (hba->capabilities & MASK_64_ADDRESSING_SUPPORT) {
 		if (!dma_set_mask_and_coherent(hba->dev, DMA_BIT_MASK(64)))
 			return 0;
 	}
-*/
 	return dma_set_mask_and_coherent(hba->dev, DMA_BIT_MASK(32));
 }
 
diff --git a/drivers/ufs/host/ufs-qcom.c b/drivers/ufs/host/ufs-qcom.c
index 69014654a645..efd29722e9eb 100644
--- a/drivers/ufs/host/ufs-qcom.c
+++ b/drivers/ufs/host/ufs-qcom.c
@@ -479,8 +479,6 @@ static int ufs_qcom_host_reset(struct ufs_hba *hba)
 
 static u32 ufs_qcom_get_hs_gear(struct ufs_hba *hba)
 {
-	return UFS_HS_G1;
-
 	struct ufs_qcom_host *host = ufshcd_get_variant(hba);
 
 	if (host->hw_ver.major == 0x1) {
@@ -1165,8 +1163,6 @@ static void ufs_qcom_advertise_quirks(struct ufs_hba *hba)
 
 static void ufs_qcom_set_caps(struct ufs_hba *hba)
 {
-	return;
-/*
 	struct ufs_qcom_host *host = ufshcd_get_variant(hba);
 
 	hba->caps |= UFSHCD_CAP_CLK_GATING | UFSHCD_CAP_HIBERN8_WITH_CLK_GATING;
@@ -1180,7 +1176,6 @@ static void ufs_qcom_set_caps(struct ufs_hba *hba)
 		host->caps = UFS_QCOM_CAP_QUNIPRO |
 			     UFS_QCOM_CAP_RETAIN_SEC_CFG_AFTER_PWR_COLLAPSE;
 	}
-*/
 }
 
 /**
@@ -1375,8 +1370,7 @@ static int ufs_qcom_common_init(struct ufs_hba *hba)
 	 * Power up the PHY using the minimum supported gear (UFS_HS_G2).
 	 * Switching to max gear will be performed during reinit if supported.
 	 */
-	host->hs_gear = UFS_HS_G1;
-	//host->hs_gear = UFS_HS_G2;
+	host->hs_gear = UFS_HS_G2;
 
 
 	return 0;
diff --git a/drivers/ufs/host/ufs-qcom.h b/drivers/ufs/host/ufs-qcom.h
index b52156c7be33..d95eb26d0cf1 100644
--- a/drivers/ufs/host/ufs-qcom.h
+++ b/drivers/ufs/host/ufs-qcom.h
@@ -27,7 +27,7 @@
 #define SLOW 1
 #define FAST 2
 
-#define UFS_QCOM_LIMIT_HS_RATE		PA_HS_MODE_A
+#define UFS_QCOM_LIMIT_HS_RATE		PA_HS_MODE_B
 
 /* QCOM UFS host controller vendor specific registers */
 enum {
diff --git a/drivers/ufs/host/ufshcd-pltfrm.c b/drivers/ufs/host/ufshcd-pltfrm.c
index 358fb9046601..0963d0bc3867 100644
--- a/drivers/ufs/host/ufshcd-pltfrm.c
+++ b/drivers/ufs/host/ufshcd-pltfrm.c
@@ -433,17 +433,17 @@ EXPORT_SYMBOL_GPL(ufshcd_get_pwr_dev_param);
 void ufshcd_init_pwr_dev_param(struct ufs_dev_params *dev_param)
 {
 	*dev_param = (struct ufs_dev_params){
-		.tx_lanes = UFS_LANE_1,
-		.rx_lanes = UFS_LANE_1,
-		.hs_rx_gear = UFS_HS_G1,
-		.hs_tx_gear = UFS_HS_G1,
-		.pwm_rx_gear = UFS_PWM_G1,
-		.pwm_tx_gear = UFS_PWM_G1,
+		.tx_lanes = UFS_LANE_2,
+		.rx_lanes = UFS_LANE_2,
+		.hs_rx_gear = UFS_HS_G3,
+		.hs_tx_gear = UFS_HS_G3,
+		.pwm_rx_gear = UFS_PWM_G4,
+		.pwm_tx_gear = UFS_PWM_G4,
 		.rx_pwr_pwm = SLOW_MODE,
 		.tx_pwr_pwm = SLOW_MODE,
 		.rx_pwr_hs = FAST_MODE,
 		.tx_pwr_hs = FAST_MODE,
-		.hs_rate = PA_HS_MODE_A,
+		.hs_rate = PA_HS_MODE_B,
 		.desired_working_mode = UFS_HS_MODE,
 	};
 }
-- 
2.47.0

